www/front_src/src/Resources/Details/Body/tabs/Details/DetailsCard/cards.tsx

```diff
@@ -0,0 +1,178 @@
+import * as React from 'react';
+
+import { Typography, Grid, makeStyles } from '@material-ui/core';
+import IconCheck from '@material-ui/icons/Check';
+
+import {
+  labelCurrentStateDuration,
+  labelPoller,
+  labelTimezone,
+  labelLastStateChange,
+  labelLastCheck,
+  labelActive,
+  labelNextCheck,
+  labelCheckDuration,
+  labelLatency,
+  labelResourceFlapping,
+  labelYes,
+  labelPercentStateChange,
+  labelLastNotification,
+  labelCurrentNotificationNumber,
+} from '../../../../../translatedLabels';
+import { getFormattedDate, getFormattedTime } from '../../../../../dateTime';
+import { ResourceDetails } from '../../../../models';
+
+interface DetailLines {
+  title: string;
+  field?: string | number | boolean;
+  lines: Array<{ key: string; line: JSX.Element | null }>;
+}
+
+const DetailsLine = ({ line }: { line?: string }): JSX.Element | null => {
+  return line ? <Typography variant="h5">{line}</Typography> : null;
+};
+
+const useStyles = makeStyles((theme) => ({
+  activeIcon: {
+    color: theme.palette.success.main,
+  },
+}));
+
+const ActiveLine = (): JSX.Element => {
+  const classes = useStyles();
+
+  return (
+    <Grid container spacing={1} alignItems="center">
+      <Grid item>
+        <IconCheck className={classes.activeIcon} />
+      </Grid>
+      <Grid item>
+        <DetailsLine key="tries" line={labelActive} />
+      </Grid>
+    </Grid>
+  );
+};
+
+const getDetailCardLines = (details: ResourceDetails): Array<DetailLines> => {
+  const getDateTimeLines = ({ label, field }): DetailLines => ({
+    title: label,
+    field,
+    lines: [
+      {
+        key: `${label}_date`,
+        line: <DetailsLine line={getFormattedDate(field)} />,
+      },
+      {
+        key: `${label}_time`,
+        line: <DetailsLine key="tries" line={getFormattedTime(field)} />,
+      },
+    ],
+  });
+
+  const getCheckLines = ({ label, field }): DetailLines => ({
+    ...getDateTimeLines({ label, field }),
+    lines: [
+      ...getDateTimeLines({ label, field }).lines,
+      {
+        key: `${label}_active`,
+        line: details.active_checks ? <ActiveLine /> : null,
+      },
+    ],
+  });
+
+  return [
+    {
+      title: labelPoller,
+      field: details.poller_name,
+      lines: [
+        {
+          key: 'poller',
+          line: <DetailsLine line={details.poller_name} />,
+        },
+      ],
+    },
+    {
+      title: labelTimezone,
+      field: details.timezone,
```

Comment:
what is the behaviour if the timezone is not defined ? empty card ?
(if I remember well, it can be null)

Reply:
The card is not displayed at all if the value is not defined (this is the standard behavior for all the cards).